core/include/optee_msg.h

```diff
@@ -204,10 +204,10 @@ struct optee_msg_arg {
 
 /*
  * Defines the maximum value of @num_params that can be passed to
- * OPTEE_MSG_GET_ARG_SIZE without wrapping the result.
+ * OPTEE_MSG_GET_ARG_SIZE without a risk of crossing page boundary.
  */
 #define OPTEE_MSG_MAX_NUM_PARAMS	\
-	((SIZE_MAX - sizeof(struct optee_msg_arg)) / \
+	((SMALL_PAGE_SIZE - sizeof(struct optee_msg_arg)) / \
```

Comment:
I'd prefer using `OPTEE_MSG_NONCONTIG_PAGE_SIZE` instead.

Reply:
Fixed and has been pushed.